src/sagemaker/session.py

```diff
@@ -3553,19 +3553,17 @@ def endpoint_from_production_variants(
         Returns:
             str: The name of the created ``Endpoint``.
         """
-        if not _deployment_entity_exists(
-            lambda: self.sagemaker_client.describe_endpoint_config(EndpointConfigName=name)
-        ):
-            config_options = {"EndpointConfigName": name, "ProductionVariants": production_variants}
-            tags = _append_project_tags(tags)
-            if tags:
-                config_options["Tags"] = tags
-            if kms_key:
-                config_options["KmsKeyId"] = kms_key
-            if data_capture_config_dict is not None:
-                config_options["DataCaptureConfig"] = data_capture_config_dict
-
-            self.sagemaker_client.create_endpoint_config(**config_options)
+        config_options = {"EndpointConfigName": name, "ProductionVariants": production_variants}
+        tags = _append_project_tags(tags)
+        if tags:
+            config_options["Tags"] = tags
+        if kms_key:
+            config_options["KmsKeyId"] = kms_key
+        if data_capture_config_dict is not None:
+            config_options["DataCaptureConfig"] = data_capture_config_dict
+
+        self.sagemaker_client.create_endpoint_config(**config_options)
+
```

Comment:
How are you checking if the endpoint exists or not
How are you notifying customers if endpoint config already exists

Reply:
your coverage is reducing. Please add unit tests

Reply:
The endpoint check happens in the create_endpoint_config call in the boto3 layer. It is raised in that method call on line 3565 and propagates up the call stack to the  endpoint_from_production_variants() method